packages/sanity/src/desk/components/pane/PaneHeader.styles.tsx

```diff
@@ -1,21 +1,42 @@
 import {Box, Flex, Layer, rgba, TextSkeleton, Text, Theme, Card} from '@sanity/ui'
 import styled, {css} from 'styled-components'
 
-export const Root = styled(Layer)`
-  line-height: 0;
-  position: sticky;
-  top: 0;
-
-  &:not([data-collapsed]):after {
-    content: '';
-    display: block;
-    position: absolute;
-    left: 0;
-    right: 0;
-    bottom: -1px;
-    border-bottom: 1px solid var(--card-shadow-outline-color);
-  }
-`
+/**
+ * Left/right border offsets for header components (in px).
+ */
+const BORDER_OFFSET_X = 12
+
+interface RootProps {
+  $borderBottom?: boolean
+  'data-collapsed'?: string
+  'data-testid'?: string
+  $isContentScrollable?: boolean
+  $hasScrolledFromTop?: boolean
+}
+
+export const Root = styled(Layer)<RootProps>(({
+  $borderBottom = true,
+  $isContentScrollable,
+  $hasScrolledFromTop,
+}) => {
+  return css`
+    line-height: 0;
+    position: sticky;
+    top: 0;
+
+    &:not([data-collapsed]):after {
+      content: '';
+      display: block;
+      position: absolute;
+      left: ${BORDER_OFFSET_X}px;
+      right: ${BORDER_OFFSET_X}px;
+      bottom: -1px;
+      border-bottom: ${$borderBottom ? '1px solid var(--card-shadow-outline-color)' : 'none'};
+      opacity: ${() => ($isContentScrollable && $hasScrolledFromTop ? 1 : 0)};
+      transition: opacity 200ms ease-in;
```

Comment:
This is close! But doesn't provide the transition that we're after –

It is applying a transition on `opacity` values, which we're varying by based on the value of `$isContentScrollable && $hasScrolledFromTop` – but we're also setting the `border-bottom` to `none`. As such, the border is 'snapping' in and out.

This could be alternatively re-written as:

```
      border-bottom: 1px solid ${$borderBottom ? 'var(--card-shadow-outline-color)' : 'transparent'};
      opacity: 1;
      transition: border-color 200ms ease-in;
```